File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for i,x in enumerate(nums):
            diff = target-x
            if diff in hashMap:
                return [i,hashMap[diff]]
            else:
                hashMap[x] = i 
        

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int)
    for i:= 0; i < len(nums);i++{
        complement := target- nums[i] 
        if j,found := seen[complement];found{
            return []int{i,j}
        }
        seen[nums[i]] = i
    }
    return nil
}